File: past-competitions/2018/src/mike-java/Prob03.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

public class Prob03 {
    // change this file name for each problem!
    private static final String INPUT_FILE_NAME = "Prob03.in.txt";
    
    public static void main(String[] args) {
        try {
            // prepare to read the file
            File inFile = new File(INPUT_FILE_NAME);
            FileReader fr = new FileReader(inFile);
            BufferedReader br = new BufferedReader(fr);
            String inLine = null;
            
            // get the number of test cases
            int T = Integer.parseInt(br.readLine());
            
            // outer loop through test cases
            while (T-- > 0) {
                // BEGINNING OF TEST CASE CODE
                
                // read the next line of text
                inLine = br.readLine();
                
                // get the number
                String[] tokens = inLine.split("th");
                int theNumber = Integer.parseInt(tokens[0]);
                
                // print the number
                System.out.print(theNumber);
                
                // check it!
                if (((theNumber % 100) == 11) || ((theNumber % 100) == 12) || ((theNumber % 100) == 13)) {
                    // numbers ending in 11, 12, and 13 are special!
                    System.out.println("th");
                } else if ((theNumber % 10) == 1) {
                    // the number has a 1 in the ones place
                    System.out.println("st");
                } else if ((theNumber % 10) == 2) {
                    // the number has a 2 in the ones place
                    System.out.println("nd");
                } else if ((theNumber % 10) == 3) {
                    // the number has a 3 in the ones place
                    System.out.println("rd");
                } else {
                    // the number has a 4-9 or a 0 in the ones place
                    System.out.println("th");
                }
                
                // END OF TEST CASE CODE
            }
            
            // clean up
            br.close();
            fr.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: past-competitions/2018/src/brett-python/Prob03.py
# Open the input file
with open("Prob03.in.txt", "rt") as inputFile:
    # Read the number of test cases (trim out the newline)
    cases = int(inputFile.readline().replace("\n", ""))
    # For each test case
    for caseNum in range(cases):
        # Read the number
        line = inputFile.readline().replace("\n", "")
        # Strip off the suffix and convert to number
        number = int(line.replace("th", ""))
        # Teens and numbers ending in 0 or 4-9 get th
        if ((number % 100) >= 10 and (number % 100) <= 20) or number % 10 == 0 or number % 10 >= 4:
            print(str(number) + "th")
        # Others get st, nd, or rd
        elif number % 10 == 1:
            print(str(number) + "st")
        elif number % 10 == 2:
            print(str(number) + "nd")
        elif number % 10 == 3:
            print(str(number) + "rd")
        
